dpppt-backend-sdk/dpppt-backend-sdk-ws/src/test/java/org/dpppt/backend/sdk/ws/controller/GaenControllerTest.java

```diff
@@ -928,40 +917,35 @@ public void zipContainsFiles() throws Exception {
 		now = UTCInstant.now();
 		var midnight = now.atStartOfDay();
 
-		// insert two times 5 keys per day for the last 14 days. the second batch has a
-		// different received at timestamp. (+6 hours)
-		insertNKeysPerDayInInterval(14,
-				midnight.minusDays(4),
-				now, now.minusDays(1));
+		// Insert two times 5 keys per day for the last 14 days. the second batch has a
+		// different received at timestamp. (+12 hours compared to the first)
+		insertNKeysPerDay(midnight, 14, 5, midnight.minusDays(1), false);
+		insertNKeysPerDay(midnight, 14, 5, midnight.minusHours(12), false);
 
-		insertNKeysPerDayInInterval(14,
-				midnight.minusDays(4),
-				now, now.minusDays(12));
-
-		// request the keys with date date 1 day ago. no publish until.
+		// request the keys with key date 8 days ago. no publish until.
 		MockHttpServletResponse response = mockMvc
-				.perform(get("/v1/gaen/exposed/"
-						+ midnight.minusDays(8).getTimestamp())
-								.header("User-Agent", "MockMVC"))
+				.perform(
+						get("/v1/gaen/exposed/" + midnight.minusDays(8).getTimestamp()).header("User-Agent", "MockMVC"))
 				.andExpect(status().is2xxSuccessful()).andReturn().getResponse();
 
 		Long publishedUntil = Long.parseLong(response.getHeader("X-PUBLISHED-UNTIL"));
 		assertTrue(publishedUntil < now.getTimestamp(), "Published until must be in the past");
 
+		// must contain 20 keys: 5 from the first insert, 5 from the second insert and
+		// 10 random keys
 		verifyZipResponse(response, 20, 144);
 
-		// request again the keys with date date 1 day ago. with publish until, so that
+		// request again the keys with date date 8 days ago. with publish until, so that
 		// we only get the second batch.
-		var bucketAfterSecondRelease = Duration.ofMillis(midnight.getTimestamp()).minusDays(1).plusHours(12).dividedBy(Duration.ofHours(2)) * 2*60*60*1000;
+		var bucketAfterSecondRelease = Duration.ofMillis(midnight.getTimestamp()).minusDays(1).plusHours(12)
+				.dividedBy(Duration.ofHours(2)) * 2 * 60 * 60 * 1000;
```

Comment:
I'm sure this can be simplified to something like:
```
var bucketAfterSecondRelease = midnight.minusHours(12);
```
as the whole division / multiplication is not needed for midnight.

And then use
```
Long.toString(bucketAfterSecondRelease.getTimeStamp())
```